jsonargparse/_postponed_annotations.py

```diff
@@ -240,6 +242,10 @@ def resolve_subtypes_forward_refs(typehint):
 
 def has_subtypes(typehint):
     typehint_origin = get_typehint_origin(typehint)
+    if typehint_origin is type:
+        if hasattr(typehint, "__args__"):
+            return True
```

Comment:
```suggestion
    if typehint_origin is type and hasattr(typehint, "__args__"):
        return True
```

Issue identified by sonarcloud.